groovy-language-server/src/main/java/com/palantir/ls/groovy/util/SourceWriter.java

```diff
@@ -69,6 +69,19 @@ public Path getDestination() {
      * Applies the given changes to the destination file. Does not handle intersecting ranges in the changes.
      */
     public synchronized void applyChanges(List<TextDocumentContentChangeEvent> contentChanges) throws IOException {
+        // Check if any of the ranges are null
+        for (TextDocumentContentChangeEvent change : contentChanges) {
+            if (change.getRange() == null) {
+                checkArgument(contentChanges.size() == 1,
+                        String.format(
+                                "Cannot have many changes when a change contains a null range which means it replaces "
+                                        + "the whole contents of the file: %s",
+                                change.toString()));
```

Comment:
It doesn't really matter which file it is though, the change is what is invalid. We also don't print out the URI below.
The thing is we won't know what the workspace root is in here (it's kind of irrelevant to the SourceWriter) and I wouldn't want to leak the absolute path.